src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputSubmittedAction.java

```diff
@@ -46,7 +47,15 @@
     public InputSubmittedAction(String approver, @CheckForNull Map<String,Object> parameters) {
         this.approver = approver;
         if (parameters != null) {
-            this.parameters.putAll(parameters);
+            for (Map.Entry<String,Object> entry : parameters.entrySet()) {
+                String k = entry.getKey();
+                Object v = entry.getValue();
+                if (v instanceof Secret) {
+                    this.parameters.put(k, "******");
```

Comment:
Why? The plaintext cannot be recovered from `build.xml` without access to global keys anyway. `PasswordParameterValue` stores a `Secret` so why should we not do the same here?